lib/Index/Index.cpp

```diff
@@ -1386,6 +1386,7 @@ bool IndexSwiftASTWalker::reportRelatedTypeRef(const TypeLoc &Ty, SymbolRoleSet
     if (auto *VD = declRefTR->getBoundDecl()) {
       if (auto *TAD = dyn_cast<TypeAliasDecl>(VD)) {
         IndexSymbol Info;
+        Info.roles |= Relations;
```

Comment:
This feels like it might be a big change but otherwise the typealias was never included as the passed relation, so the index entry was just a reference, but never more specific like it's the base of the type etc

Reply:
Hmmm. I don't feel super strongly about this, but it does feel pretty odd to me. Given:
```
struct A {}
typealias SomeAlias = A
extension SomeAlias {}
```

This would add a `RelationExtendedBy` on `SomeAlias` (and on the implicit reference `A`). That feels a little odd to me. Did you need it to be more specific?

Reply:
I think that example is actually correct? In that case it looks like this:

```swift
/*
       constructor=init()=s:3foo1AVACycfc=definition, implicit, childOf
       struct=A=s:3foo1AV=definition
       v  */
struct A {}
/*
          typealias=SomeAlias=s:3foo9SomeAliasa=definition
          |           struct=A=s:3foo1AV=reference
          v           v  */
typealias SomeAlias = A
/*
          typealias=SomeAlias=s:3foo9SomeAliasa=reference, extendedBy
          struct=A=s:3foo1AV=reference, implicit, extendedBy
          extension.swiftExtensionOfStruct=A=s:e:s:3foo1AVAAyyF=definition
          v  */
extension SomeAlias {
    /*
         instanceMethod=foo()=s:3foo1AVAAyyF=definition, childOf
         v  */
    func foo() {}
}
```

And since you can technically call `foo()` on an `A` it seems fair that it is implicitly extended by? As in because this compiles:

```swift
struct A {}
typealias SomeAlias = A
extension SomeAlias {
    func foo() {}
}

let a = A()
a.foo()
```

Reply:
Oh sorry, yea you mean the other way around, the typealias being `extendedBy`. That one is a bit weirder I agree. Conceptually it does seem to me like we are extending the typealias, but I guess practically speaking that's not really happening. I think the baseOf case is more clear, especially if you wanted to try to build a graph of these to trace things back. This isn't a hugely core part of this change so we could drop it if you want.

Reply:
My preference would be that `A` gets an implicit `extendedBy` and `SomeAlias` just gets a reference, especially if you don't actually have a use case in mind where you need `SomeAlias` to be `extendedBy` at the moment.

Reply:
sounds good, done